lib/puppet/provider/iis_application/webadministration.rb

```diff
@@ -0,0 +1,129 @@
+    fail "Error destroying application: #{result[:errormessage]}" unless result[:errormessage].nil?
+  end
+
+  def update
+    check_paths
+    inst_cmd = []
+
+    inst_cmd << "$webApplication = Get-WebApplication -Site '#{resource[:site]}' -Name '#{resource[:applicationname]}'"
+    if @property_flush[:physicalpath]
+      #XXX Under what conditions would we have other paths?
+      inst_cmd << %{Set-WebConfigurationProperty -Filter "$($webApplication.ItemXPath)/virtualDirectory[@path='/']" -Name physicalPath -Value '#{@resource[:physicalpath]}' -ErrorAction Stop}
+    end
+
+    if @property_flush[:sslflags]
+      flags = @property_flush[:sslflags].join(',')
+      inst_cmd << "Set-WebConfigurationProperty -Location '#{@resource[:site]}/#{@resource[:applicationname]}' -Filter 'system.webserver/security/access' -Name 'sslFlags' -Value '#{flags}' -ErrorAction Stop"
+    end
+
+    if @property_flush[:authenticationinfo]
+      @property_flush[:authenticationinfo].each do |auth,enable|
+        inst_cmd << "Set-WebConfigurationProperty -Location '#{@resource[:site]}/#{@resource[:applicationname]}' -Filter 'system.webserver/security/authentication/#{auth}Authentication' -Name enabled -Value #{@property_flush[:authenticationinfo][auth]} -ErrorAction Stop"
+      end
+    end
+
+    inst_cmd = inst_cmd.join("\n")
+    result   = self.class.run(inst_cmd)
+    fail "Error updating application: #{result[:errormessage]}" unless result[:exitcode] == 0
+    fail "Error updating application: #{result[:errormessage]}" unless result[:errormessage].nil?
+  end
+
+  def self.prefetch(resources)
+    apps = instances
+    resources.each do |name, resource|
+      if provider = apps.find{ |app| app.site == resource[:site] && app.applicationname == resource[:applicationname] }
+        resources[name].provider = provider
+      end
+    end
+  end
+
+  def self.instances
+    inst_cmd = ps_script_content('getapps', @resource)
+    result   = run(inst_cmd)
+    text     = result[:stdout]
+    return [] if text.nil?
+
+    app_json = JSON.parse(text)
+    app_json = [app_json] if app_json.is_a?(Hash)
+    app_json.collect do |app|
+      app_hash = {}
+
+      app_hash[:ensure]             = :present
+      app_hash[:name]               = "#{app['site']}\\#{app['name']}"
+      app_hash[:applicationname]    = app['name']
+      app_hash[:site]               = app['site']
+      app_hash[:physicalpath]       = app['physicalpath']
+      app_hash[:applicationpool]    = app['applicationpool']
+      app_hash[:sslflags]           = app['sslflags']
+      app_hash[:authenticationinfo] = app['authenticationinfo']
+
+      new(app_hash)
+    end
+  end
+
+  private
+
+  def check_paths
+    if @resource[:physicalpath] and ! File.exists?(@resource[:physicalpath])
+      fail "physicalpath doesn't exist: #{@resource[:physicalpath]}"
+    end
+    #XXX How do I check for IIS:\ path existence without shelling out to PS?
```

Comment:
You will have to go to PS, since IIS:\ is a PSProvider. Alternately you can go down to the raw xml. IMHO, go the PowerShell route